tool/tsh/common/app.go

```diff
@@ -518,11 +518,15 @@ func getAppInfo(cf *CLIConf, tc *client.TeleportClient, matchRouteToApp func(tls
 	}
 
 	// If we didn't find an active profile for the app, get info from server.
-	app, err := getApp(cf.Context, tc, cf.AppName)
+	app, logins, err := getApp(cf.Context, tc, cf.AppName)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	if len(logins) == 0 {
+		logins = getARNFromRoles(cf, tc, profile, app)
+	}
```

Comment:
> should this also be deleted in 18.0?

The idea is that once we delete the `getARNFromRoles`, we also drop this if statement. If that needs to be clarified, I can also add the deletion note here.

> len(logins) == 0 could be a legit scenario where the user just doesn't have any AWS roles assigned. Is it possible to differentiate the two different scenarios?

No. The `Login` field is already on the resource and wasn't being filled out for apps. But, if that happens, the final result will be the same (because the access checker will return empty).

Reply:
> But, if that happens, the final result will be the same (because the access checker will return empty).

Extra API alls though. But i think it would rare.